lib/ansible/modules/apt.py

```diff
@@ -964,6 +975,19 @@ def cleanup(m, purge=False, force=False, operation=None,
     m.exit_json(changed=changed, stdout=out, stderr=err, diff=diff)
 
 
+def aptclean(m):
+    clean_rc, clean_out, clean_err = m.run_command(['apt-get', 'clean'])
+    if m._diff:
+        clean_diff = parse_diff(clean_out)
+    else:
+        clean_diff = {}
+    if clean_rc:
+        m.fail_json(msg="apt-get clean failed", stdout=clean_out, rc=clean_rc)
+    if clean_err:
+        m.fail_json(msg="apt-get clean failed: %s" % clean_err, stdout=clean_out, rc=clean_rc)
+    m.exit_json(changed=True, msg=clean_out, stdout=clean_out, stderr=clean_err, diff=clean_diff)
```

Comment:
should be 'additive' to result when other operations are also executed, unless you force to make it an exclusive argument to all others

Reply:
I'm not quite sure what you mean by additive. Can you clarify?

Reply:
Hm, I had modeled this after the upgrade() function since it should work in a similar manner. I'm not sure what results dictionary you're referring to. Can you point me to a similar example that outlines the logic you're describing?

Reply:
The result of this:
`- apt: clean=yes package=whatever state=installed`

would be that 'whatever' is ignored and user gets told 'all ok and changed', this is misleading. while 'upgrade' goes through many hoops to ensure compatibility with others states/info (too many IMHO, i would simplify this a lot)

Reply:
That is very helpful, thank you. Let me do some research and testing and I'll follow up on this.

Reply:
@bcoca Digging into the logic and running some tests, I noticed that when using "update_cache=yes" along with "install=...", the update_cache won't return a changed state and defers to whatever install reports. If I remove the final m.exit_json() in the aptclean function, it will work the same way. Would that be an acceptable way to handle this if it was documented appropriately?

Reply:
Latest commit applies this logic. Please let me know what you think.

Reply:
@bcoca Logic has been updated so that now:

- Running clean=yes by itself will return the correct changed status and exit
- Running clean=yes with a package install/update/removal will defer status change to the state (matches update_cache logic)
- Any errors encountered will exit and print the error 

When you have time, can you look over the latest updates and provide feedback? Thank you very much!